src/systems/weapons/player_weapons/player_blaster/spawn_player_blaster.rs

```diff
@@ -13,17 +16,32 @@ use bevy::{
     math::Vec3,
     prelude::{AssetServer, AudioBundle, Commands, KeyCode, Query, Res, ResMut, Transform, With},
     sprite::{Sprite, SpriteBundle},
+    time::Time,
     utils::tracing,
 };
 
+//TODO Find proper fix for this
+#[allow(clippy::too_many_arguments)]
```

Comment:
Need to look at some kind of refactor for the spawn weapon funcs as clippy is complaining about too many params. Maybe similar fix to queries where we have a stuct of the params?

Reply:
Maybe consider a refactor at this point for spawning a player blaster as they do a lot in this function